rpcs3/Emu/RSX/Common/ProgramStateCache.h

```diff
@@ -599,6 +549,16 @@ class program_state_cache
 
 	void clear()
 	{
+		std::scoped_lock(m_vertex_mutex, m_fragment_mutex, m_decompiler_mutex, m_pipeline_mutex);
```

Comment:
this is a bug because you do not assign an object.
unless this is what you meant to lock and then immediatly unlock, in this case use shared_mutex::lock_unlock().

Reply:
Oh right, yea. Kinda looks like `std::scoped lock(xxx)`